mkosi/__init__.py

```diff
@@ -1064,6 +1064,65 @@ def secure_boot_sign(state: MkosiState, directory: Path, replace: bool = False)
                 os.rename(f"{f}.signed", f)
 
 
+def secure_boot_configure_auto_enroll(state: MkosiState) -> None:
+    if state.do_run_build_script:
+        return
+    if not state.config.bootable:
+        return
+    if not state.config.secure_boot:
+        return
+    if state.for_cache:
+        return
+
+    with complete_step("Setting up secure boot auto-enrollment…"):
+        keys_dir = state.root / "boot/loader/keys/auto"
+        keys_dir.mkdir(parents=True, exist_ok=True)
+
+        # sbsiglist expects a DER certificate.
+        run(
+            [
+                "openssl",
+                "x509",
+                "-outform",
+                "DER",
+                "-in",
+                state.config.secure_boot_certificate,
+                "-out",
+                keys_dir / "mkosi.der",
```

Comment:
Do we need the `der` in the final image? If not, you can put it in `state.workspace` instead so it'll get auto cleaned up

Reply:
Nope. Same with the esl. Good to know I can put them there.